website/identifiers/clients/datacite.py

```diff
@@ -129,44 +130,54 @@ def build_doi(self, object):
     def get_identifier(self, identifier):
         self._client.doi_get(identifier)
 
-    def create_identifier(self, node, category):
-        if category == 'doi':
-            metadata = self.build_metadata(node)
-            if settings.DATACITE_ENABLED:
-                resp = self._client.metadata_post(metadata)
-                # Typical response: 'OK (10.70102/FK2osf.io/cq695)' to doi 10.70102/FK2osf.io/cq695
-                doi = re.match(r'OK \((?P<doi>[a-zA-Z0-9 .\/]{0,})\)', resp).groupdict()['doi']
-                self._client.doi_post(doi, node.absolute_url)
-                return {'doi': doi, 'metadata': metadata}
-            logger.info('TEST ENV: DOI built but not minted')
-            return {'doi': self.build_doi(node), 'metadata': metadata}
-        else:
+    def create_identifier(self, node, category, doi_value=None):
+        if category != 'doi':
             raise NotImplementedError('Creating an identifier with category {} is not supported'.format(category))
 
-    def update_identifier(self, node, category):
-        if settings.DATACITE_ENABLED and not node.is_public or node.is_deleted:
-            if category == 'doi':
-                doi = self.build_doi(node)
-                self._client.metadata_delete(doi)
-                return {'doi': doi}
-            else:
-                raise NotImplementedError('Updating metadata not supported for {}'.format(category))
+        doi_value = doi_value or node.get_identifier_value('doi') or self.build_doi(node)
+        metadata = self.build_metadata(node, doi_value=doi_value)
+        if settings.DATACITE_ENABLED:
+            resp = self._client.metadata_post(metadata)
+            # Typical response: 'OK (10.70102/FK2osf.io/cq695)' to doi 10.70102/FK2osf.io/cq695
+            doi = re.match(r'OK \((?P<doi>[a-zA-Z0-9 .\/]{0,})\)', resp).groupdict()['doi']
+            self._client.doi_post(doi, node.absolute_url)
+            return {'doi': doi, 'metadata': metadata}
         else:
-            return self.create_identifier(node, category)
+            logger.info('TEST ENV: DOI built but not minted')
+
+        return {'doi': doi_value, 'metadata': metadata}
```

Comment:
nit: it would be nice to have consistency with the crossref client with these return values.